qemu/tests/nic_hotplug.py

```diff
@@ -238,14 +262,18 @@ def device_add_nic(pci_model, netdev, device_id):
                 s_session = vm.reboot(session=s_session, serial=True)
                 vm.verify_alive()
             s_session.close()
-    finally:
-        for nic in nic_hotplugged:
-            vm.hotunplug_nic(nic.nic_name)
+
         logging.info("Re-enabling the primary link(s)")
         for nic in primary_nic:
             vm.set_link(nic.device_id, up=True)
-        logging.info("Reboot vm to verify it alive after hotunplug nic(s)")
-        serial = not (len(vm.virtnet) > 0)
-        session = vm.reboot(serial=serial)
-        vm.verify_alive()
-        session.close()
+
+        plug_tag = "after_plug"
+        vm_switched_off = run_sub_test(params, plug_tag)
+        if vm_switched_off:
+            return
+
+        for nic in nic_hotplugged:
+            vm.hotunplug_nic(nic.nic_name)
+
+        plug_tag = "after_unplug"
+        run_sub_test(params, plug_tag)
```

Comment:
@kimi1978wy  I prefer to add this step description in line 30 , maybe adding 10). It will make the description more accurate and easier to read.